File: Leetcode First 200 JS solution/56. Merge Intervals.js
var merge = function(itv) {
    
    itv.sort((a, b) => a[0] - b[0]);
    
    if (itv.length < 1) return itv;
    
    let ans = [];
    ans.push(itv[0])
    
    for (let i = 1; i < itv.length; i++) {
        
        if (ans[ans.length-1][1] < itv[i][0]){
            ans.push(itv[i]);
        } else {
            let value = ans.pop();
            const front = Math.min(value[0], itv[i][0]);
            const end   = Math.max(value[1], itv[i][1]);
            ans.push([front, end]);
        }
    }
    return ans;
};

// o(n) solution
// 

File: Leetcode First 200 Python solution/56.Merge Intervals.py
# Merge Intervals
class Solution:
	
	def merge(self, intervals):
		
		intervals.sort()
        
		ans = [intervals[0]]
		
		for interv in intervals[1:]:
			
			if interv[0] > ans[-1][1]:
				ans.append(interv)
				continue
			
			up_interv = ans.pop()
			
			front_val = min(up_interv[0], interv[0])
			back_val = min(up_interv[1], interv[1])
			
			ans.append([front_val, back_val])
			
		return ans

# o(n) solution here!
# o(n) Time complexity solution here!

	